scm/driver/bitbucket/git.go

```diff
@@ -28,13 +29,25 @@ func (s *gitService) CreateBranch(ctx context.Context, repo string, params *scm.
 }
 
 func (s *gitService) FindBranch(ctx context.Context, repo, name string) (*scm.Reference, *scm.Response, error) {
+	if strings.Contains(name, "/") {
+		path := fmt.Sprintf("/2.0/repositories/%s/refs/branches?q=name=\"%s\"", repo, name)
+		out := new(branches)
+		res, err := s.client.do(ctx, "GET", path, nil, out)
+		return convertBranch(out.Values[0]), res, err
+	}
 	path := fmt.Sprintf("2.0/repositories/%s/refs/branches/%s", repo, name)
 	out := new(branch)
 	res, err := s.client.do(ctx, "GET", path, nil, out)
 	return convertBranch(out), res, err
 }
 
 func (s *gitService) FindCommit(ctx context.Context, repo, ref string) (*scm.Commit, *scm.Response, error) {
+	if strings.Contains(ref, "/") {
+		path := fmt.Sprintf("2.0/repositories/%s/refs/branches?q=name=\"%s\"", repo, ref)
+		out := new(branches)
+		res, err := s.client.do(ctx, "GET", path, nil, out)
+		return convertCommit(out.Values[0].Target), res, err
```

Comment:
if there is an error, `out.Values[0]` would result in an index out of bounds exception and a panic.  Also are we sure the first branch in the list is going to be the correct branch?

Reply:
If the branch Name is wrong then this will be nulltr. I can wrap the error and do the name check?

Reply:
The branch list is a name exact match so there should be exactly one. I will add error handling if it is not 1